src/client/proxy/http.cc

```diff
@@ -144,7 +148,14 @@ void HTTPProxyHandler::AsyncHandleReadHeaders(
         boost::asio::buffers_end(buf));
     // add the additional bytes the m_body
     m_Protocol.m_Body = str;
-  }
+  } else if (num_additional_bytes < 0) {
```

Comment:
`num_additional_bytes` is *unsigned*. Research the difference between signed and unsigned types (and overflow + underflow for both types) to learn why this line makes no sense. Even if this type was signed (but it's not so don't get caught up with what I'm about to say), you don't want to test this way; you'll usually want to throw a class from std::exception.